src/components/DayPickerSingleDateController.jsx

```diff
@@ -0,0 +1,572 @@
+    });
+  }
+
+  onDayMouseLeave() {
+    const { hoverDate, visibleDays } = this.state;
+    if (this.isTouchDevice || !hoverDate) return;
+
+    const modifiers = this.deleteModifier({}, hoverDate, 'hovered');
+
+    this.setState({
+      hoverDate: null,
+      visibleDays: {
+        ...visibleDays,
+        ...modifiers,
+      },
+    });
+  }
+
+  onPrevMonthClick() {
+    const { onPrevMonthClick, numberOfMonths, enableOutsideDays } = this.props;
+    const { currentMonth, visibleDays } = this.state;
+
+    const newVisibleDays = {};
+    Object.keys(visibleDays).sort().slice(0, numberOfMonths + 1).forEach((month) => {
+      newVisibleDays[month] = visibleDays[month];
+    });
+
+    const prevMonth = currentMonth.clone().subtract(1, 'month');
+    const prevMonthVisibleDays = getVisibleDays(prevMonth, 1, enableOutsideDays);
+
+    this.setState({
+      currentMonth: prevMonth,
+      visibleDays: {
+        ...newVisibleDays,
+        ...this.getModifiers(prevMonthVisibleDays),
+      },
+    });
+
+    onPrevMonthClick();
+  }
+
+  onNextMonthClick() {
+    const { onNextMonthClick, numberOfMonths, enableOutsideDays } = this.props;
+    const { currentMonth, visibleDays } = this.state;
+
+    const newVisibleDays = {};
+    Object.keys(visibleDays).sort().slice(1).forEach((month) => {
+      newVisibleDays[month] = visibleDays[month];
+    });
+
+    const nextMonth = currentMonth.clone().add(numberOfMonths, 'month');
+    const nextMonthVisibleDays = getVisibleDays(nextMonth, 1, enableOutsideDays);
+
+    this.setState({
+      currentMonth: currentMonth.clone().add(1, 'month'),
+      visibleDays: {
+        ...newVisibleDays,
+        ...this.getModifiers(nextMonthVisibleDays),
+      },
+    });
+
+    onNextMonthClick();
+  }
+
+
+  getFirstFocusableDay(newMonth) {
+    const { date, numberOfMonths } = this.props;
+
+    let focusedDate = newMonth.clone().startOf('month');
+    if (date) {
+      focusedDate = date.clone();
+    }
+
+    if (this.isBlocked(focusedDate)) {
+      const days = [];
+      const lastVisibleDay = newMonth.clone().add(numberOfMonths - 1, 'months').endOf('month');
+      let currentDay = focusedDate.clone();
+      while (!isAfterDay(currentDay, lastVisibleDay)) {
+        currentDay = currentDay.clone().add(1, 'day');
+        days.push(currentDay);
+      }
+
+      const viableDays = days.filter(day => !this.isBlocked(day) && isAfterDay(day, focusedDate));
+      if (viableDays.length > 0) focusedDate = viableDays[0];
+    }
+
+    return focusedDate;
+  }
+
+  getModifiers(visibleDays) {
+    const modifiers = {};
+    Object.keys(visibleDays).forEach((month) => {
+      modifiers[month] = {};
+      visibleDays[month].forEach((day) => {
+        modifiers[month][toISODateString(day)] = this.getModifiersForDay(day);
+      });
+    });
+
+    return modifiers;
+  }
+
+  getModifiersForDay(day) {
+    return new Set(Object.keys(this.modifiers).filter(modifier => this.modifiers[modifier](day)));
```

Comment:
should this be `new Set(Object.values(this.modifiers).map(x => x(day)))`? (using `object.values` ofc)

(fine as-is, too)